### PR TITLE
Universal self assembling gun part kit

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds the universal self assembling gun parts kit. Basically just an item that can be used on a gun kit to craft the gun, rather than having to get the specific gun required to craft it.

## Why It's Good For The Game

Infinite gun printing is bad. However, antags being able to use the guns from time to time is good, so long as it is not _free._

As such, this PR gives you the option to buy your way to an RND gun, at a steep price, rather than breaking into the armory / robbing an officer.

Can be lowered to 6 TC, but doing 8 TC unless balance wants it at 6, of which I would gladly oblige, but 8 is a good starting point. After all, you can get 2 stechkins for that price, or a syndicate teleporter to get into the armory... either of which is probably a better deal, but some people might have gimics in mind with the more exotic weapons, or perhaps wants a carbine because security has a mech, wants to go pyro man with an immolator, wants to go shock guardian with arc revolver, ect, ect.

## Images of changes

image

## Testing

Confirmed in uplink / worked

## Changelog
:cl:
add: Universal self assembling gun part kit, antagonists can purchase it for 8 tc to complete a gunkit without a gun.
/:cl:

### CODE REVIEW

File: code/modules/crafting/guncrafting.dm [hunk 1]
```diff
@@ -24,66 +24,102 @@
...
+		playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+		if(!do_after(user, 5 SECONDS, target = user))
+			return
+		playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+		var/obj/item/I = new G.outcome
+		user.unEquip(src)
+		user.put_in_hands(I)
+		qdel(G)
+		qdel(src)
+		return
```
Comment (GDNgit) [inline code/modules/crafting/guncrafting.dm:122]:
```suggestion
```
dangling return

File: code/modules/crafting/guncrafting.dm [hunk 2]
```diff
@@ -24,66 +24,101 @@
...
 	desc = "A suitcase containing the necessary gun parts to transform a standard disabler into a silenced and lethal disabling weapon. Look officer, he has no wounds from me!"
 	origin_tech = "combat=6;magnets=6;syndicate=2"
+	outcome = /obj/item/gun/energy/disabler/silencer
+
+/obj/item/weaponcrafting/gunkit/universal_gun_kit
+	name = "\improper universal self assembling gun parts kit"
+	desc = "A suitcase containing the necessary gun parts to build a full gun, when combined with a gun kit. Use it directly on a gunkit to rapidly assemble it."
+	icon_state = "syndicase"
+
+/obj/item/weaponcrafting/gunkit/universal_gun_kit/afterattack(atom/movable/AM, mob/user, flag)
```
Comment (GDNgit) [inline code/modules/crafting/guncrafting.dm:107]:
```suggestion
/obj/item/weaponcrafting/gunkit/universal_gun_kit/afterattack(obj/item/weaponcrafting/gunkit/gunkit_to_use, mob/user, flag)
	if(!istype(gunkit_to_use))
		return
	if(!gunkit_to_use.outcome)
		to_chat(user, "<span class='warning'>That gunkit can not be used to craft a weapon.</span>")
		return
	playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
	if(!do_after(user, 5 SECONDS, target = user))
		return
	playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
	var/obj/item/gun_produced = new gunkit_to_use.outcome
	user.unEquip(src)
	user.put_in_hands(gun_produced)
	qdel(gunkit_to_use)
	qdel(src)
```

File: code/modules/crafting/guncrafting.dm [hunk 3]
```diff
@@ -24,66 +24,101 @@
...
+			return
+		playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+		if(!do_after(user, 5 SECONDS, target = user))
+			return
+		playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+		var/obj/item/I = new G.outcome
+		user.unEquip(src)
+		user.put_in_hands(I)
+		qdel(G)
+		qdel(src)
```
Comment (GDNgit) [inline code/modules/crafting/guncrafting.dm:121]: Changes suggested:
More descriptive var name(s), guard clause, general simplification of assignment code
```suggestion
/obj/item/weaponcrafting/gunkit/universal_gun_kit/afterattack(obj/item/weaponcrafting/gunkit/gunkit_to_use, mob/user, flag)
	if(!istype(gunkit_to_use))
		return
	if(!gunkit_to_use.outcome)
		to_chat(user, "<span class='warning'>That gunkit can not be used to craft a weapon.</span>")
		return
	playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
	if(!do_after(user, 5 SECONDS, target = user))
		return
	playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
	var/obj/item/gun_produced = new gunkit_to_use.outcome
	user.unEquip(src)
	user.put_in_hands(gun_produced)
	qdel(gunkit_to_use)
	qdel(src)
```

**Comment by McRamon** [comment]
Why is it science restricted? I dont see really why it should be job specific

**Comment by Qwertytoforty** [comment]
> Why is it science restricted? I dont see really why it should be job specific

It was made job specific as it is harder for other antagonists to make use of it, / not sure how people would feel about every job getting universal gun kits. It can be made for all jobs if people think it is a good idea

**Comment by Carthusia** [comment]
Im not a huge fan of it being job limited.

There are few items that really should be job limited, like the wooloo staff, his grace, but for the rest, it should just be up to the players to decide how they wana run their antag role, not a random job selection chance, combined with a random antag selection chance.

**Comment by Qwertytoforty** [comment]
I'll remove it from being job restricted then

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**